### PR TITLE
Adding weather events

### PR DESCRIPTION
**Pull request by Daniel-Glz**

### CODE REVIEW

**Comment by connerdouglass** [comment]
Other than my one small comment, I think this looks great. Maybe @AL1L can review it also and then we'll merge

File: src/events/types/weather/LightningStrikeEvent.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    
+    /*
+    * Gets the bolt which is striking the earth
+    * @returns 
+    */
+        
+    /*
+    ---------- LightningStrike entity needs to be added -----------
+
+    public getLightning():LightningStrike {
```
Comment (AL1L) [inline src/events/types/weather/LightningStrikeEvent.ts]: You can just return an Entity for now. 

Although as the Entity type has only one extra method, I'd prefer you go ahead and make the Entity class, don't have to though.

File: src/events/types/weather/LightningStrikeEventCause.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export enum LightningStrikeEventCause {
+    COMMAND = 'COMMAND',
```
Comment (AL1L) [inline src/events/types/weather/LightningStrikeEventCause.ts:6]: This file needs documentation

File: src/events/types/weather/LightningStrikeEvent.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * Stores data for lightning striking
+ */
+export class LightningStrikeEvent extends WeatherEvent implements Cancellable {
+    
+    public static getBukkitClasspath(): string {
+        return 'org.bukkit.event.weather.LightningStrikeEvent';
+    }
+   
+    /**
+    * Gets the cause of this lightning strike
```
Comment (AL1L) [inline src/events/types/weather/LightningStrikeEvent.ts]: Almost all of your method documentation is missing one space, the `*`s should all line up. I guess it's not that important, but would be nice to fix if you have the time

File: src/events/types/weather/LightningStrikeEvent.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    * @returns strike cause
+    */
+    public getCause():LightningStrikeEventCause {
```
Comment (AL1L) [inline src/events/types/weather/LightningStrikeEvent.ts]: Missing a space before the return type.

@connerdouglass I was thinking, we should add eslint and prettier

File: src/events/types/weather/LightningStrikeEvent.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        return this.toJava().getCause();
```
Comment (AL1L) [inline src/events/types/weather/LightningStrikeEvent.ts]: This should have `.getNane()` to get the enum name.

**Review by AL1L — CHANGES_REQUESTED** [review]

**Comment by connerdouglass** [comment]
Today I made some changes on `master` to the entity classes. Please merge your branch with the latest changes on master, and use the `LightningStrike` entity I added there, instead of yours. Thanks!

**Comment by Daniel-Glz** [comment]
> Today I made some changes on `master` to the entity classes. Please merge your branch with the latest changes on master, and use the `LightningStrike` entity I added there, instead of yours. Thanks!

Ready, my branch is up to date with the latest changes.

**Pull request merged**